cmd/gateway/azure/gateway-azure.go

```diff
@@ -675,52 +689,57 @@ func getAzureMetadataObjectName(objectName, uploadID string) string {
 	return fmt.Sprintf(metadataObjectNameTemplate, uploadID, sha256.Sum256([]byte(objectName)))
 }
 
-func (a *azureObjects) checkUploadIDExists(bucketName, objectName, uploadID string) (err error) {
+func (a *azureObjects) checkUploadIDExists(ctx context.Context, bucketName, objectName, uploadID string) (err error) {
 	blob := a.client.GetContainerReference(bucketName).GetBlobReference(
 		getAzureMetadataObjectName(objectName, uploadID))
 	err = blob.GetMetadata(nil)
-	err = azureToObjectError(errors.Trace(err), bucketName, objectName)
+	logger.LogIf(ctx, err)
+	err = azureToObjectError(err, bucketName, objectName)
 	oerr := minio.ObjectNotFound{
 		Bucket: bucketName,
 		Object: objectName,
 	}
 	if errors.Cause(err) == oerr {
-		err = errors.Trace(minio.InvalidUploadID{
+		logger.LogIf(ctx, minio.InvalidUploadID{UploadID: uploadID})
+		err = minio.InvalidUploadID{
 			UploadID: uploadID,
-		})
+		}
 	}
 	return err
 }
 
 // NewMultipartUpload - Use Azure equivalent CreateBlockBlob.
 func (a *azureObjects) NewMultipartUpload(ctx context.Context, bucket, object string, metadata map[string]string) (uploadID string, err error) {
 	uploadID = mustGetAzureUploadID()
-	if err = a.checkUploadIDExists(bucket, object, uploadID); err == nil {
-		return "", errors.Trace(fmt.Errorf("Upload ID name collision"))
+	if err = a.checkUploadIDExists(ctx, bucket, object, uploadID); err == nil {
+		logger.LogIf(ctx, fmt.Errorf("Upload ID name collision"))
```

Comment:
We can remove the logging here since the error condition is taken care of inside `a.checkUploadIDExists`.